homeassistant/components/slack/notify.py

```diff
@@ -193,14 +193,22 @@ async def _async_send_remote_file_message(
 
     async def _async_send_text_only_message(self, targets, message, title, blocks):
         """Send a text-only message."""
+        username = self._username
+        icon = self._icon
```

Comment:
Why is this variable needed? It's only used in one place below.

Reply:
Because of the dereferencing of the dictionary needed for the dynamic parameter name, the context of `self` becomes the slack message client. Setting the class variable to a local variable allows a proper closure to be formed when passing it to `chat_postMessage()`